tfhe/src/core_crypto/algorithms/lwe_encryption.rs

```diff
@@ -2264,16 +2832,396 @@ pub fn par_encrypt_lwe_compact_ciphertext_list_with_compact_public_key<
+/// let mut random_generator = RandomGenerator::<ActivatedRandomGenerator>::new(seeder.seed());
+///
+/// let crs = CompactPkeCrs::new(
+///     lwe_dimension,
+///     lwe_ciphertext_count.0,
+///     glwe_noise_distribution,
+///     ciphertext_modulus,
+///     plaintext_modulus,
+///     &mut random_generator,
+/// )
+/// .unwrap();
+///
+/// // Create the LweSecretKey
+/// let lwe_secret_key =
+///     allocate_and_generate_new_binary_lwe_secret_key(lwe_dimension, &mut secret_generator);
+///
+/// let lwe_compact_public_key = allocate_and_generate_new_lwe_compact_public_key(
+///     &lwe_secret_key,
+///     glwe_noise_distribution,
+///     ciphertext_modulus,
+///     &mut encryption_generator,
+/// );
+///
+/// let cleartexts = (0..lwe_ciphertext_count.0 as u64).collect::<Vec<_>>();
+///
+/// // Create a new LweCompactCiphertextList
+/// let mut output_compact_ct_list = LweCompactCiphertextList::new(
+///     0u64,
+///     lwe_dimension.to_lwe_size(),
+///     lwe_ciphertext_count,
+///     ciphertext_modulus,
+/// );
+///
+/// let proof = par_encrypt_and_prove_lwe_compact_ciphertext_list_with_compact_public_key(
+///     &lwe_compact_public_key,
+///     &mut output_compact_ct_list,
+///     &cleartexts,
+///     delta,
+///     glwe_noise_distribution,
+///     glwe_noise_distribution,
+///     &mut secret_generator,
+///     &mut encryption_generator,
+///     &mut random_generator,
+///     crs.public_params(),
+///     ZkComputeLoad::Proof,
+/// )
+/// .unwrap();
+///
+/// // verify the ciphertext list with the proof
+/// assert!(verify_lwe_compact_ciphertext_list(
+///     &output_compact_ct_list,
+///     &lwe_compact_public_key,
+///     &proof,
+///     crs.public_params(),
+/// )
+/// .is_valid());
+///
+/// let mut output_plaintext_list =
+///     PlaintextList::new(0u64, PlaintextCount(lwe_ciphertext_count.0));
+///
+/// let lwe_ciphertext_list = output_compact_ct_list.expand_into_lwe_ciphertext_list();
+///
+/// decrypt_lwe_ciphertext_list(
+///     &lwe_secret_key,
+///     &lwe_ciphertext_list,
+///     &mut output_plaintext_list,
+/// );
+///
+/// let signed_decomposer =
+///     SignedDecomposer::new(DecompositionBaseLog(4), DecompositionLevelCount(1));
+///
+/// // Round the plaintexts
+/// output_plaintext_list
+///     .iter_mut()
+///     .for_each(|x| *x.0 = signed_decomposer.closest_representable(*x.0) >> 60);
+///
+/// // Check we recovered the original messages
+/// assert_eq!(&cleartexts, output_plaintext_list.as_ref());
+/// ```
+#[cfg(feature = "zk-pok-experimental")]
+#[allow(clippy::too_many_arguments)]
+pub fn par_encrypt_and_prove_lwe_compact_ciphertext_list_with_compact_public_key<
+    Scalar,
+    KeyCont,
+    InputCont,
+    OutputCont,
+    MaskDistribution,
+    NoiseDistribution,
+    SecretGen,
+    EncryptionGen,
+    G,
+>(
+    lwe_compact_public_key: &LweCompactPublicKey<KeyCont>,
+    output: &mut LweCompactCiphertextList<OutputCont>,
+    messages: &InputCont,
```

Comment:
missed that during review

https://github.com/zama-ai/tfhe-rs-internal/issues/520

don't want to break things now, too late to make small fixes, but I think we should stick to plaintexts

Reply:
or whatever type + encoding information